### PR TITLE
add types for hsl-to-rgb-for-reals

### PR DESCRIPTION
**Pull request by jorenbroekema**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by gqio — APPROVED** [review]

File: types/hsl-to-rgb-for-reals/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+// Project: https://github.com/davidmarkclements/hsl_rgb_converter/
+// Definitions by: jorenbroekema <https://github.com/DefinitelyTyped>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/**
+ * Convert hue, saturation and luminosity to rgb
+ *
+ * e.g. hslToRgb(223, 0.44, 0.56) to [93, 121, 192]
+ */
+declare function hslToRgb(hue: number, saturation: number, lightness: number): [number, number, number];
```
Comment (peterblazejewicz) [inline types/hsl-to-rgb-for-reals/index.d.ts:11]:
```suggestion
declare function hslToRgb(hue: number, saturation: number, lightness: number): [red: number, green: number, blue: number];
```
you can safely add named tuples (and this make sense for RGB), as TS4.0 is now minimal version support in DT ecosystem:
https://github.com/DefinitelyTyped/DefinitelyTyped#support-window

File: types/hsl-to-rgb-for-reals/hsl-to-rgb-for-reals-tests.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import hslToRgb = require('hsl-to-rgb-for-reals');
+
+hslToRgb(223, 0.44, 0.56); // $ExpectType [number, number, number]
```
Comment (peterblazejewicz) [inline types/hsl-to-rgb-for-reals/hsl-to-rgb-for-reals-tests.ts:3]:
```suggestion
hslToRgb(223, 0.44, 0.56); // $ExpectType [red: number, green: number, blue: number]
```
(if you accept named tuple change)

File: types/hsl-to-rgb-for-reals/index.d.ts (hunk 1 above)
Comment (jorenbroekema) [reply types/hsl-to-rgb-for-reals/index.d.ts:11]: Ah cool, that's better yeah

**Review by gqio — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]
@jorenbroekema thanks!

**Comment by jorenbroekema** [comment]
Ready to merge

**Pull request merged**